siafolder.go

```diff
@@ -527,17 +506,14 @@ func (sf *SiaFolder) removeDeleted() error {
 }
 
 // filters Sia remote files, only files that match prefix parameter are returned
-func (sf *SiaFolder) getSiaFiles() (rf api.RenterFiles, err error) {
-	renterFiles, err := sf.client.RenterFilesGet(true)
+func (sf *SiaFolder) getSiaFiles() (map[modules.SiaPath]modules.FileInfo, error) {
+	siaSyncDir, err := sf.client.RenterGetDir(getSiaPath(prefix))
 	if err != nil {
-		return rf, err
+		return nil, err
 	}
-	for _, siafile := range renterFiles.Files {
-		// we should only match prefix directory in sia and nothing else
-		if !strings.HasPrefix(siafile.SiaPath.Path, prefix+"/") {
-			continue
-		}
-		rf.Files = append(rf.Files, siafile)
+	siaFiles := make(map[modules.SiaPath]modules.FileInfo)
+	for _, file := range siaSyncDir.Files {
```

Comment:
I'm not sure this works, if the files never existed on Sia previously doesn't this give a `no such file or directory` error?

```
tbenz9@spartan1:~/go/src/github.com/tbenz9/siasync$ ./siasync -debug -subfolder test2 /tmp/siasync
2019/09/26 16:50:17 Connected to Sia  1.4.1.2-master
2019/09/26 16:50:17 30  contracts are ready for upload
2019/09/26 16:50:17 [DEBUG] Calculating checksum for: /tmp/siasync/new-folder/file.txt
2019/09/26 16:50:17 [DEBUG] Calculating checksum for: /tmp/siasync/new-folder3/README.md
2019/09/26 16:50:17 [DEBUG] Calculating checksum for: /tmp/siasync/new-folder4/CHANGELOG.md
2019/09/26 16:50:17 Uploading files missing from Sia
2019/09/26 16:50:17 [GET request error; failed to get directory contents:open /home/tbenz9/Documents/Sia/renter/siafiles/test2/.siadir: no such file or directory]
tbenz9@spartan1:~/go/src/github.com/tbenz9/siasync$
```

You can recreate it by pointing Siasync at a directory it's never tracked before.  like `siasync -subfolder test /tmp/a/new/directory/`